### PR TITLE
few more arm architectures in builtin.zig

### PR DESCRIPTION
**Pull request by shawnl**

### CODE REVIEW

**Comment by andrewrk** [comment]
@vegecode can you sanity check this?

**Comment by vegecode** [comment]
I think the warning is unnecessary. The memcpy that compiler-rt uses is the bulitin zig version and I'm assuming llvm will produce correct code that doesn't have unaligned memory accesses.

As for the thumb 1 architectures, it is true that those architectures support thumb instructions, but they have to be explicitly told to generate thumb code with -mthumb as a command line option. My project that required this is for an armv7em device which only supports thumb2 and no arm instructions at all so effectively -mthumb is always set, if you will.

I did not do much testing of code generation comparison for the other architectures. There was nothing there before so any addition was an improvement. Now that I am looking back into it and comparing code generation with different options, `use_thumb_1` should only be true on some of the items already listed if it is using thumb code generation.

tldr; It is not strictly correct right now, but merging the new architectures is just as wrong as the others. We can make an issue and I can fix them all at once. There is no test like actually running it, though. Maybe I can borrow old dev boards with mcus that have those architectures from work.

**Comment by shawnl** [comment]
> As for the thumb 1 architectures, it is true that those architectures support thumb instructions, but they have to be explicitly told to generate thumb code with -mthumb as a command line option.

Ahh, on Debian the toolchain handles this for you. Debian's `armel` port supports armv4t+, so that is why I knew that we could do this.

**Comment by vegecode** [comment]
Well the v4t can do thumb 1 or traditional arm instructions, even both at
the same time with each function specifying what mode the processor needs
to be in to decode the instructions within that function.

I need to figure out what mode the aeabi_memcpy call is generated in to
determine which mode it should be. It's entirely possible that the compiler
produces arm mode function calls on the older architectures because they
aren't as developed on the llvm toolchain.

GCC was the compiler of choice until very recently when Arm switched their
compiler to a clang one with custom extensions.

The assembly can also be misleading because the assembler can easily
translate those push and pop instructions (which are thumb) to the arm
equivalent. I just need to investigate more

**Comment by vegecode** [comment]
Also I'm in the middle of a move to another house so I'll be unavailable to do that kind of investigating pretty much for a week or so.

**Comment by andrewrk** [comment]
I think this was solved by @vegecode in a different PR
